File: sms_back_end/database/multi_tenant_school_management/schemas/student.py
from typing import Optional, List
from pydantic import BaseModel
from uuid import UUID

from .parent_student_light import ParentBasic
from .user import UserCreate, UserRead


class StudentBase(BaseModel):
    admission_number: str
    class_id: Optional[UUID]


class StudentCreate(StudentBase):
    user: UserCreate


class StudentRead(StudentBase):
    id: UUID
    user: UserRead
    model_config = {"from_attributes": True}


class StudentAdmissionNumberResponse(BaseModel):
    admission_number: str

class StudentWithParents(StudentRead):
    parents: List[ParentBasic] = []

File: sms-frontend/types/student.ts
import {UserCreate, UserRead} from "@/types/user";
import {UUID} from "@/types/shared.primitive";

export interface StudentBase {
    admission_number: string;
    class_id?: UUID | null;
}

export interface StudentCreate extends StudentBase {
    user: UserCreate;
}

export interface StudentRead extends StudentBase {
    id: UUID;
    user: UserRead;
}

export interface ParentBasic {
    id: UUID;
}

export interface StudentWithParents extends StudentRead {
    parents: ParentBasic[];
}